tools/PI/DevHome.PI/Controls/AddToolControl.xaml.cs

```diff
@@ -101,55 +161,290 @@ private void ClearValues()
             return null;
         }
 
-        var argType = ExternalToolArgType.None;
-
-        if (HwndRadio.IsChecked ?? false)
+        var activationType = ToolActivationType.Launch;
+        if (ProtocolRadio.IsChecked ?? false)
         {
-            argType = ExternalToolArgType.Hwnd;
+            activationType = ToolActivationType.Protocol;
         }
-        else if (ProcessIdRadio.IsChecked ?? false)
+        else if (selectedApp is not null && selectedApp.IsMsix)
         {
-            argType = ExternalToolArgType.ProcessId;
+            activationType = ToolActivationType.Msix;
         }
 
         return new(
             ToolNameTextBox.Text,
             ToolPathTextBox.Text,
-            argType,
-            PrefixTextBox.Text ?? string.Empty,
-            OtherArgsTextBox.Text ?? string.Empty,
+            activationType,
+            ArgumentsTextBox.Text,
+            selectedApp?.AppUserModelId ?? string.Empty,
+            selectedApp?.IconFilePath ?? string.Empty,
             IsPinnedToggleSwitch.IsOn);
     }
 
-    private void UpdateSampleCommandline(object sender, TextChangedEventArgs e)
+    private void CancelButton_Click(object sender, RoutedEventArgs e)
     {
-        UpdateSampleCommandline();
+        ClearValues();
     }
 
-    private void UpdateSampleCommandline(object sender, RoutedEventArgs e)
+    private async void AppListButton_Click(object sender, RoutedEventArgs e)
     {
-        UpdateSampleCommandline();
+        allApps.Clear();
+        SortedApps.Clear();
+
+        itemCount = GetShortcuts();
+        if (itemCount == 0)
+        {
+            Log.Error("Error getting shortcuts");
```

Comment:
Is the error reason logged prior to this part in the code?

Reply:
GetShortcuts is a very simple function, diagnosis would be trivial.